parser.y

```diff
@@ -1479,6 +1546,14 @@ AlterTableSpec:
 			Num:             getUint64FromNUM($6),
 		}
 	}
+|	"ALTER" "PARTITION" Identifier PlacementSpecList %prec lowerThanComma
+	{
+		$$ = &ast.AlterTableSpec{
+			Tp:             ast.AlterTableAlterPartition,
+			PartitionNames: []model.CIStr{model.NewCIStr($3)},
+			PlacementSpecs: $4.([]*ast.PlacementSpec),
```

Comment:
If placement rules on partitions are also stored in `PlacementSpecs`, how can you figure out whether they belong to the table or the partition?
E.g.
```sql
ALTER TABLE t
        ALTER PLACEMENT ..., 
        ALTER PARTITION ALTER PLACEMENT ...
```

Reply:
`Tp = AlterTableAlterPartition` for `ALTER PARTITION ALTER PLACEMENT...`, and `Tp = AlterTablePlacement` for `ALTER PLACEMENT...`. And they should belong to two different `AlterTableSpec`. I think that is enough.